guides/debugging.md

```diff
@@ -0,0 +1,44 @@
+Did you ever see yourself placing `ShowDebug` everywhere in the code with comments like `Been here`, `If ok` and etc, just for trying to understand why your new groundbreaking feature isn't working correctly? The solution is called breakpoints.
+
+## What is debugging?
+Simply put, debugging is the ability to put breakpoints in your code and when the server will take that given path, it will pause the rest of the execution, and once it's paused you can inspect variables values, evaluate expressions directly on the `Immediate Window` and so on.
```

Comment:
Does the period after `given path` really makes sense? English is not my mother tongue but in Portuguese that would break the semantics of the sentence I think

Reply:
Does the period after given path really makes sense? English is not my mother tongue but in Portuguese that would break the semantics of the sentence I think

Reply:
Good catch! I was trying to fix the run on sentence and messed it up. I've fixed the original suggestion, let me know if that sounds better!